RIOT-OS/RIOT
# nimble/netif: add support for BT5 PHY modes

### Contribution description
This PR adds support for the additional BLE PHY modes that were introduced with Bluetooth 5, the 2Mbit mode as well as the Coded (long range) PHY  mode. 

To use these modes, extended advertisements are required by the Bluetooth standard. Once this mode is enabled in NimBLE, NimBLE switches to a different set of GAP API calls for scanning and connecting. To keep the `nimble_netif` API independent form these underlying API switches, two NImBLE-independent configuration structs used by the `nimble_netif_connect()` and `nimble_netif_accept()` are introduced. This way `nimble_netif` can transparently used with or without enabling extended advertising mode (`MYNEWT_VAL_BLE_EXT_ADV`).

These small API changes to those 2 functions have the further advantage, that the API can be extended in the future without breaking it, again :-)

This PR is rebased on a bunch of fixes and foremost on the extended `nimble_scanner` PR, as the entirety of these changes is useful for testing and they all play together...


### Testing procedure
Run e.g. `gnrc_networking` on two boards with IP over BLE support. Use the build-in shell commands (`ble adv `, `ble_adv_ext`, `ble_adv_dir`) to advertise in different modes, while using the `ble scan` shell command to verify the selected modes. Also establish connections between the nodes using different PHY modes, e.g. `ble connect nodename 1000 CODED`

NOTE: the Coded (long-range) PHY mode is not supported by the `nrf52832` line of CPUs...

- gnrc_networking (for regression bugs)
- tests/nimble_netif_ext
- tests/nimble_rpble_gnrc + rpble_ext
- tests/nimble_statconn_grnc + statconn_ext
- tests/nimble_autoconn_gnrc + autoconn_ext

### Issues/PRs references
rebased on top of ~~#16838, #16839, #16843, and #16859~~ -> #16838, #16839, #16859 have been merged.
EDIT: all dependencies have been merged.
